pennylane_lightning_kokkos/src/util/GetConfigInfo.hpp

```diff
@@ -0,0 +1,180 @@
+// Copyright 2023 Xanadu Quantum Technologies Inc.
+
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+
+//     http://www.apache.org/licenses/LICENSE-2.0
+
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+/**
+ * @file GetConfigInfo.hpp
+ * Defines operations to store information from Kokkos::print_configuration in
+ * a unordered_map variable.
+ */
+#pragma once
+#include "Kokkos_Core.hpp"
+#include <cstring>
+#include <sstream>
+#include <string>
+#include <unordered_map>
+#include <utility>
+#include <vector>
+namespace {
+/*@brief Split a string into a string vector based on the delimiters.
+ *
+ *@param str String to be splitted.
+ *@param delimiters Pointer to the delimiters.
+ *@return str_list Substring vector.
+ * */
+auto string_split(const std::string &str, const char *delimiters)
+    -> std::vector<std::string> {
+    std::vector<std::string> str_list;
+    char *token = std::strtok(const_cast<char *>(str.data()), delimiters);
+    while (token != 0) {
+        str_list.push_back(token);
+        token = std::strtok(nullptr, delimiters);
+    }
+    return str_list;
+}
+
+/*@brief Check if a string contains a substring.
+ *
+ *@param substr The substring to be checked.
+ *@param string The string to be checked.
+ *@return bool type.
+ * */
+bool is_substr(const std::string &substr, const std::string &str) {
+
+    if (str.find(substr) != std::string::npos)
+        return true;
+    else
+        return false;
+}
+/*@brief Parse the backend details based on the output of
+ *Kokkos::print_configuration.
+ *
+ *@return meta_map std::unordered_map<std::string,
+ *std::unordered_map<std::string, std::string>> type.
+ * */
+auto getConfig() {
+
+    std::ostringstream buffer;
+
+    Kokkos::print_configuration(buffer, true);
+
+    const std::string bufferstr(buffer.str());
+
+    const auto str_list = string_split(bufferstr, "\n");
+
+    std::vector<std::string> query_keys;
+
+    using value_map = std::unordered_map<std::string, std::string>;
+
+    using category_map = std::unordered_map<std::string, value_map>;
+
+    std::size_t num_devices = 0;
+
+    category_map meta_map;
+
+    for (std::size_t i = 0; i < str_list.size(); i++) {
```

Comment:
Let me try to if we can reduce the number of branches here. The main reason why there are many branches here is because different backends print backend info in different ways. Good idea to add  a sample googlebenchmark example .

Reply:
Here is the benchmark result. It takes 15 microseconds after reducing number of branches, which is 25% faster than previous implementation. 
<img width="565" alt="image" src="https://user-images.githubusercontent.com/31480676/216690225-94152cb6-6a00-436c-a939-e27444649f4d.png">

Reply:
Thanks Shuli. I think we can probably not worry much about this for future given the load runtime, though for future readability we may want to deal directly with the `ExecSpaceManager` and define our own output directly. Let's worry about this at a later date. Thanks for the updates.